### PR TITLE
Add Engineblock Stepup configuration

### PR DESCRIPTION
**Pull request by pablothedude**
We need to set the default Stepup configuration to handle 2fa
authentication through the Stepup Gateway.

### CODE REVIEW

**Comment by pablothedude** [comment]
@thijskh could you validate if sane defaults are being used?

File: roles/engineblock/defaults/main.yml [hunk 1]
```diff
@@ -64,9 +68,21 @@ engine_minimum_execution_time_on_invalid_received_response: 5000
 engine_time_frame_for_authentication_loop_in_seconds: 60
 engine_maximum_authentication_procedures_allowed: 5
 
-# Regex used to blacklist AuthnContextClassRef attributes from IDP's
-engine_sfo_authn_context_class_ref_blacklist_regex: ''
-
+# This PCRE regex is used to blacklist incoming AuthnContextClassRef attributes on. If an empty string is used
+# the validation is skipped. The validator will throw an exception if the used regex is invalid.
+engine_stepup_authn_context_class_ref_blacklist_regex: ''
```
Comment (thijskh) [inline roles/engineblock/defaults/main.yml]: Might want to give an example regex, e.g. In a comment

Comment (pablothedude) [reply roles/engineblock/defaults/main.yml]: With the templating engine in Ansible I'm able to generate a sane default based on the Stepup domain used. Will that be a good solution?

`engine_stepup_authn_context_class_ref_blacklist_regex: '/http:\/\/{{ engine_stepup_base_domain | regex_escape }}\/assurance\/loa[1-3]/'`

File: roles/engineblock/defaults/main.yml [hunk 2]
```diff
@@ -64,9 +68,21 @@ engine_minimum_execution_time_on_invalid_received_response: 5000
...
-
+# This PCRE regex is used to blacklist incoming AuthnContextClassRef attributes on. If an empty string is used
+# the validation is skipped. The validator will throw an exception if the used regex is invalid.
+engine_stepup_authn_context_class_ref_blacklist_regex: '/http:\/\/{{ engine_stepup_base_domain | regex_escape }}\/assurance\/loa[1-3]/'
+# The loa mapping from the internal used LoA's to the Stepup Gateway LOA's
+engine_stepup_engineblock_loa1: "http://{{ base_domain }}/assurance/loa1"
+engine_stepup_engineblock_loa2: "http://{{ base_domain }}/assurance/loa2"
+engine_stepup_engineblock_loa3: "http://{{ base_domain }}/assurance/loa3"
+engine_stepup_gateway_loa2: "http://{{ engine_stepup_base_domain }}assurance/loa2"
+engine_stepup_gateway_loa3: "http://{{ engine_stepup_base_domain }}assurance/loa3"
```
Comment (thijskh) [inline roles/engineblock/defaults/main.yml]: These two lines miss a `/` between domain and the word “assurance”

File: roles/engineblock/defaults/main.yml [hunk 3]
```diff
@@ -38,6 +38,10 @@ engine_debug: false
 
 engine_profile_baseurl: "https://profile.{{ base_domain }}"
 
+# Required for Stepup authentication
+engine_stepup_base_domain: "stepup.example.com"
```
Comment (thijskh) [inline roles/engineblock/defaults/main.yml]: Could `example.com` also instead be `{{ base_domain }}`?

File: roles/engineblock/defaults/main.yml (hunk 2 above)
Comment (pablothedude) [reply roles/engineblock/defaults/main.yml]: Done

File: roles/engineblock/defaults/main.yml (hunk 3 above)
Comment (pablothedude) [reply roles/engineblock/defaults/main.yml]: Done

**Pull request merged**